x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/xpack/ssl/SSLReloadDuringStartupIntegTests.java

```diff
@@ -59,66 +81,121 @@ protected boolean transportSSLEnabled() {
-                new Thread(() -> {
-                    latch.countDown();
-                    try {
-                        latch.await();
-                        Thread.sleep(randomLongBetween(1L, 2000L));
-                    } catch (InterruptedException e) {
-                        Thread.currentThread().interrupt();
-                    }
-
-                    // replace the bad keystore with a new copy of the good one
+                LOGGER.info("Node [{}] stopped in {}ms.", nodeName, TimeValue.timeValueNanos(System.nanoTime() - stopNanos).millisFrac());
+                atomicCopyIfPossible(badKeystorePath, nodeKeystorePath);
+                final Thread fixKeystoreThread = new Thread(() -> {
+                    waitUntilNodeStartupIsReadyToBegin(beforeKeystoreFix); // SYNC: Cert update & ES restart
                     try {
-                        copyAndAtomicMoveIfPossible(origKeystorePath, keystorePath);
-                        writtenLatch.countDown();
+                        atomicCopyIfPossible(goodKeystorePath, nodeKeystorePath);
+                        afterKeystoreFix.countDown(); // SYNC: Cert update & ES restart
                     } catch (IOException e) {
                         throw new UncheckedIOException(e);
                     }
-                }).start();
-
-                latch.countDown();
-                try {
-                    latch.await();
-                } catch (InterruptedException e) {
-                    Thread.currentThread().interrupt();
-                }
-                return super.onNodeStopped(nodeName);
+                });
+                fixKeystoreThread.setName("Fix Keystore");
+                fixKeystoreThread.start();
+                waitUntilFixKeystoreIsReadyToBegin(beforeKeystoreFix); // SYNC: Cert update & ES restart
+                return super.onNodeStopped(nodeName); // ASSUME: RestartCallback will do ES start next
             }
         });
-        writtenLatch.await();
-        ensureClusterSizeConsistency();
-        ensureFullyConnectedCluster();
+        try (Timed t = new Timed(LOGGER, Level.INFO, "Awaited {}ms. Verifying the cluster...")) {
+            afterKeystoreFix.await(); // SYNC: Verify cluster after cert update
+        }
+        try (Timed t = new Timed(LOGGER, Level.INFO, "Ensure cluster size consistency took {}ms.")) {
+            ensureClusterSizeConsistency();
+        }
+        try (Timed t = new Timed(LOGGER, Level.INFO, "Ensure fully connected cluster took {}ms.")) {
+            ensureFullyConnectedCluster();
+        }
+    }
+
+    private void waitUntilNodeStartupIsReadyToBegin(final CountDownLatch beforeKeystoreFix) {
+        beforeKeystoreFix.countDown(); // SYNC: Cert update & ES restart
+        try {
+            final long sleepMillis = randomLongBetween(1L, 2000L); // intended sleepMillis
+            try (Timed t = new Timed(LOGGER, Level.INFO, "Awaited {}ms. Sleeping " + sleepMillis + "ms before fixing...")) {
```

Comment:
can you move all of these timers down to TRACE or DEBUG level and then bump the logging for the test via the `@TestLogging` annotation

Reply:
Done